app/services/network_topology_service.rb

```diff
@@ -0,0 +1,124 @@
+class NetworkTopologyService < TopologyService
+  def initialize(provider_id)
+    @provider_id = provider_id
+    @providers = retrieve_providers(ManageIQ::Providers::NetworkManager, @provider_id)
+    # @entity = retrieve_entity(entity_id, entity_type)
+    @entity = @providers.first
+  end
+
+  def entity_type(entity)
+    if entity.kind_of?(CloudNetwork)
+      entity.class.base_class.name.demodulize
+    else
+      entity.class.name.demodulize
+    end
+  end
+
+  def build_topology
+    topo_items = {}
+    links = []
+
+    entity_relationships = {
+      :NetworkManager => {
```

Comment:
are you basing your PR on the recursive topology PR? that PR is wip, it will take some time till it will be merged.